src/txorphanage.cpp

```diff
@@ -135,7 +135,8 @@ unsigned int TxOrphanage::LimitOrphans(unsigned int max_orphans)
         EraseTx(m_orphan_list[randompos]->first);
         ++nEvicted;
     }
-    return nEvicted;
+    if (nEvicted > 0) LogPrint(BCLog::MEMPOOL, "orphanage overflow, removed %u tx\n", nEvicted);
+    return nExpired + nEvicted;
```

Comment:
The return value isn't used outside of tests, so I think it can be removed. If someone is interested in the size, they can just call `.size()`.

Reply:
```cpp
auto size_before = orphanage.Size();
auto limit = fuzzed_data_provider.ConsumeIntegral<unsigned int>();
auto n_evicted = WITH_LOCK(g_cs_orphans, return orphanage.LimitOrphans(limit));
Assert(size_before - n_evicted <= limit);
Assert(orphanage.Size() <= limit);
```
In most cases `size_before - n_evicted == orphange.Size()`, so we don't need the return value of `LimitOrphans`. But IMO it will not hurt to use the return value for an additional check. Actually I also think `EraseForBlock` can return `nErased`.

Reply:
> In most cases size_before - n_evicted == orphange.Size(), so we don't need the return value of LimitOrphans

Can you give an example where this wouldn't hold, assuming *after* the changes in this pull requests?

Reply:
The equality wouldn't hold if `n_evicted` or `n_expired` is incremented but `EraseTx` fail to remove tx from `m_orphanage`. It seems this should work normally since the `txid` passed to `EraseTx` is valid

Reply:
Yeah, I am thinking, either:

* the equality holds, in which case the return value is unused and redundant, also confuse the devs which is the "correct" way to get the size. Also, if the return value was used in the future, it can trivially be added back.
* the equality doesn't hold, in which case the return value is confusing, since it is wrong.

Regardless of which case holds, removing the return value is likely the correct "fix".